src/main.rs

```diff
@@ -64,11 +97,55 @@ fn main() {
     }
 }
 
+async fn lake_logger(
+    stats: Arc<Mutex<Stats>>,
+    view_client: actix::Addr<near_client::ViewClientActor>,
+) {
+    let interval_secs = 10;
+    let mut prev_done_count: u64 = 0;
+
+    loop {
+        tokio::time::sleep(std::time::Duration::from_secs(interval_secs)).await;
+        let stats_lock = stats.lock().await;
+        let stats_copy = stats_lock.clone();
+        drop(stats_lock);
+
+        let block_processing_speed: f64 =
+            (stats_copy.done_count as f64 - prev_done_count as f64) / interval_secs as f64;
```

Comment:
I'd prefer to add braces here and in all places where you do math + casts. 
It could divide by `interval_secs` without changing the type, and then cast the result to `f64`.
I don't remember the parsing order, it's better not to rely on this

Reply:
I hope I get it right, could you have a look again?

Reply:
You got it right, now it's better!